ProteoformSuiteInternal/Proteoform.cs

```diff
@@ -52,62 +49,139 @@ public List<ExperimentalProteoform> identify_connected_experimentals()
 
                 double mass_tolerance = this.modified_mass / 1000000 * (double)Lollipop.mass_tolerance;
                 int sign = Math.Sign(e.modified_mass - modified_mass);
-                double deltaM = Math.Sign(r.peak_center_deltaM) < 0 ? r.peak_center_deltaM : sign * r.peak_center_deltaM; // give EE relations the correct sign, but don't switch negative ET relation deltaM's
-                ModificationWithMass best_addition = null;
+                double deltaM = Math.Sign(r.peak.peak_deltaM_average) < 0 ? r.peak.peak_deltaM_average : sign * r.peak.peak_deltaM_average; // give EE relations the correct sign, but don't switch negative ET relation deltaM's
+                TheoreticalProteoform theoretical_base = this as TheoreticalProteoform != null ?
+                    this as TheoreticalProteoform : //Theoretical starting point
+                    (linked_proteoform_references.First.Value as TheoreticalProteoform != null ?
+                        linked_proteoform_references.First.Value as TheoreticalProteoform : //Experimental with theoretical reference
+                        null); //Experimental without theoretical reference
```

Comment:
Starting with theoretical proteoforms, experimental proteoforms are identified in layers (breadth-first search). Therefore, the first proteoform in the linked list is the theoretical proteoform, and then the next identified experimental proteoform, and so on. If it is already assigned, the proteoform isn't added to the list and nothing about the identification is changed (the end of the search).